lib/beaker/dsl/helpers/puppet_helpers.rb

```diff
@@ -300,8 +300,18 @@ def puppet_conf_for host, conf_opts
           new_conf
         end
 
+        # Restarts the named puppet service
+        #
+        # @param [Host] host Host the service runs on
+        # @param [String] service Name of the service to restart
+        # @param [Fixnum] curl_retries Number of times to retry the restart command
+        # @param [Fixnum] port Port to check status at
+        #
+        # @return [Result] Result of last status check
         # @!visibility private
-        def bounce_service host, service, curl_retries = 120
+        def bounce_service host, service, curl_retries = nil, port = nil
+          curl_retries = 120 if curl_retries.nil?
```

Comment:
Hm.  I think just leaving this as `curl_retries = nil` will do the defaulting, unless the desired use-case is:

``` ruby
bounce_service node, 'fooservice', nil, 1234
```

i.e., specifying `nil` in order to get the default value of `curl_retries` when specifying `port`.

I'm not super :smile_cat: on that approach -- it's how hash arguments were invented, basically.  I'm willing to sign off on it here, as this is the interface we 

For the wider (not this PR) conversation, I wonder if we should be thinking about kwargs (keyword args) for our various parameters. The downside there would be interface change (not the end of the world, especially if it improves the interface). Depends on ruby version, and I'm not 100% clear on what we expect for ruby version.